nixos/modules/services/networking/go-autoconfig.nix

```diff
@@ -0,0 +1,98 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.go-autoconfig;
+  configFile = cfg: pkgs.writeText "go-autoconfig.conf" ''
+    service_addr: ":${cfg.port}"
+
+    domain: ${cfg.domain}
+
+    imap:
+      server: ${cfg.imap.server}
+      port: ${cfg.imap.port}
+
+    smtp:
+      server: ${cfg.smtp.server}
+      port: ${cfg.smtp.port}
+  '';
```

Comment:
is this yaml? if so, `pkgs.formats.yaml` would be better (and would allow adding an RFC42 `settings` option)